services/QuillLMS/app/services/learn_worlds_integration/sso_request.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module LearnWorldsIntegration
+  class SSORequest < UserRequest
+    class NilUserError < StandardError; end
+    class NilEmailError < StandardError; end
+
+    attr_reader :user
+
+    delegate :email, :learn_worlds_account, :username, to: :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    def run
+      raise NilUserError if user.nil?
+      raise NilEmailError if email.nil?
+
+      HTTParty.post(endpoint, body: body, headers: headers)
```

Comment:
I'm pretty sure that newer Ruby syntactic sugar allows you to implicitly pull values if they are stored in a variable with the same name as the key.  So I think you can write this as `HTTParty.post(endpoint, body:, headers:)` which avoids a bit of unnecessary boilderplate.